WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsSiteProvider.kt

```diff
@@ -0,0 +1,14 @@
+package org.wordpress.android.ui.bloggingprompts
+
+import androidx.lifecycle.SavedStateHandle
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import javax.inject.Inject
+
+class BloggingPromptsSiteProvider @Inject constructor(
+    private val savedStateHandle: SavedStateHandle
```

Comment:
I haven't tried `SavedStateHandle` yet so I might be wrong: should we be using [getLiveData](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate#savedstate-livedata) instead? 🤔

Reply:
My reasoning was that it's only being used for analytics events in the view model. So livedata seemed unnecessary.

Does that make sense?

Reply:
Yes, makes sense 👍  thanks for clarifying

Reply:
Last question about this: is this part of [the docs](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate#setup) relevant to this scenario?

> When providing a custom [ViewModelProvider.Factory](https://developer.android.com/reference/androidx/lifecycle/ViewModelProvider.Factory) instance, you can enable usage of SavedStateHandle by extending [AbstractSavedStateViewModelFactory](https://developer.android.com/reference/androidx/lifecycle/AbstractSavedStateViewModelFactory).

Reply:
Yes but I believe Hilt is taking care of it internally. I ran the code to make sure it's running.